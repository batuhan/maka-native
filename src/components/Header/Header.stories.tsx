// src/components/Header/Header.stories.tsx

import React from "react";
import { Text } from "react-native";
import { storiesOf } from "@storybook/react-native";

import Header from "./Header";

storiesOf("Header", module).add("Header", () => (
  <Header title="Header Title">
    <Text>Header Logo</Text>
  </Header>
));
