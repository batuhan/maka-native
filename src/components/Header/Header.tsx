import React, { ReactNode } from "react";
import { View, Text } from "react-native";
import styled from "styled-components";

interface Props {
  style?: {};
  children: ReactNode;
  title: string;
}

export default function HeaderComponent({ title, children, style }: Props) {
  return (
    <View style={style}>
      {children}
      <Text>{title}</Text>
    </View>
  );
}
