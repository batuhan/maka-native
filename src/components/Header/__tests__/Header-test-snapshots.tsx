import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";

import Header from "../Header";

describe("Header", () => {
  it("renders correctly", () => {
    const title = "Header Title";
    const instance = renderer.create(
      <Header style={{}} title={title}>
        <Text>Header Logo</Text>
      </Header>
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
