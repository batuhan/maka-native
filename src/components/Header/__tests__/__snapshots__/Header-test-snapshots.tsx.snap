// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={Object {}}
>
  <Text>
    Header Logo
  </Text>
  <Text>
    Header Title
  </Text>
</View>
`;
